tests/test_model_saving.py

```diff
@@ -232,5 +232,35 @@ def test_loading_weights_by_name_2():
     assert_allclose(np.zeros_like(jessica[1]), jessica[1])  # biases init to 0
 
 
+# a function to be called from the Lambda layer
+def square_fn(x):
+    return x + x
```

Comment:
Good catch, it had a different name first. I changed it to `x * x` now. Defining it inside the test function does not work, because that would give the function a `__closure__` that can not be serialized.

That's actually another problem: I think [this line](https://github.com/fchollet/keras/blob/edae1785327dd7a418ac06c2fe85a8c1f6ea05b7/keras/utils/generic_utils.py#L42-L43) responsible for serializing lambda functions is not correct. Saving only the `cell_contents` might work for serializing, but it gives problems during deserialization when [these lines](https://github.com/fchollet/keras/blob/edae1785327dd7a418ac06c2fe85a8c1f6ea05b7/keras/utils/generic_utils.py#L56-L59) expect the full cell objects and not just the `cell_contents`.